### PR TITLE
Improve text extraction API

### PR DESCRIPTION
**Pull request by pinaraf**
Extracting the text from the whole page is a nice feature, but the pdfium API makes it possible to extract pieces of the page instead.
Expose the TextRect API (and working around its dangerous calls), with associated unit tests.
Cf. #4 and sorry for the delay...

### CODE REVIEW

File: src/pdfium/qpdfiumpage.cpp [hunk 1]
```diff
@@ -109,18 +111,45 @@ QImage QPdfiumPage::image(qreal scale)
...
 
     return image;
+}
+
+int QPdfiumPage::countChars() const
+{
+    return m_pageHolder->m_textPage->CountChars();
+}
 
+std::vector<QRectF> QPdfiumPage::getTextRects(int start, int count) const
```
Comment (paulovap) [inline src/pdfium/qpdfiumpage.cpp]: Hi. Do you mind change the return to QList<QRectF> or QVector<QRectF>?  I think its better to use Qt containers at the moment.

**Comment by paulovap** [comment]
The change looks very good! I just commented a small change that might be necessary. I also will start a small documentation (qdoc) so if you have time I would be great to document the methods you created. But its not mandatory so I will merge it if you don't have time.

File: src/pdfium/qpdfiumpage.cpp (hunk 1 above)
Comment (pinaraf) [inline src/pdfium/qpdfiumpage.cpp]: Hi

I'll do it, no problem.
Regarding the doc, I'll let you setup the skeleton and I will then document my functions.
I may also commit a small sample application I built using these functions, it's still in progress…

**Comment by pinaraf** [comment]
I did the change you requested. Need anything else ?

**Comment by paulovap** [comment]
Looks great! Thank you for your time.

**Pull request merged**